### PR TITLE
Add fields to discoverd PVs.

### LINKED ISSUE #134: Need additional fields during PV discovery
Currently the reported pv objects look like this when written back to the MigPlan:

```
  persistentVolumes:
  - name: pv1
    supportedActions:
    - copy
    - move
```

The UI requires a number of additional fields for us to render the PV tables as shown in the mocks, the following fields are missing:

`namespace` - the namespace the PV was found in
`size` - size of PV
`claim` - associated pvc name
`storageClass` - associated storageClass, although we'll probably need to think about this more in depth given the issues that Dylan raised around differing storage classes in two different clusters.

### PR DESCRIPTION
**Pull request by jortel**
fixes #134

Add requested fields:
- PVC that associates the Pod to the PV.
- Capacity
- Size

### CODE REVIEW

File: config/crds/migration_v1alpha1_migplan.yaml [hunk 1]
```diff
@@ -28,6 +28,8 @@ spec:
           type: object
         spec:
           properties:
+            closed:
+              type: boolean
```
Comment (jortel) [inline config/crds/migration_v1alpha1_migplan.yaml]: Generated by `make manifests`. This seems to be missing on _master_.

File: pkg/apis/migration/v1alpha1/migplan_types.go [hunk 2]
```diff
@@ -530,22 +532,33 @@ const (
...
+	PVC              PVC               `json:"pvc,omitempty"`
 	staged           bool
 }
 
+// PVC
+type PVC types.NamespacedName
+
 // Update the PV with another.
-func (r *PersistentVolume) Update(pv PersistentVolume) {
+func (r *PV) Update(pv PV) {
```
Comment (jortel) [inline pkg/apis/migration/v1alpha1/migplan_types.go]: Renamed to use well know acronym.

File: pkg/apis/migration/v1alpha1/migplan_types.go [hunk 3]
```diff
@@ -530,22 +532,33 @@ const (
...
 // staged - A PV has been explicitly added/updated.
-type PersistentVolume struct {
-	Name             string   `json:"name,omitempty"`
-	StorageClass     string   `json:"storageClass,omitempty"`
-	SupportedActions []string `json:"supportedActions"`
-	Action           string   `json:"action,omitempty"`
+type PV struct {
+	Name             string            `json:"name,omitempty"`
+	Capacity         resource.Quantity `json:"capacity,omitempty"`
+	Size             int               `json:"size,omitempty"`
```
[the remark this answers is not in the document]
Comment (djwhatle) [reply pkg/apis/migration/v1alpha1/migplan_types.go]: I tested this, it produces output that looks like this:
```
  spec:
    [...]
    namespaces:
    - mysql-persistent
    persistentVolumes:
    - capacity: 100Gi
      name: pv5
      pvc:
        Name: mysql
        Namespace: mysql-persistent
      size: 221
      supportedActions:
      - copy
      - move
```

I feel like the mix of `capacity: 100Gi` (has units) and `size: 221` (no units) is kind of strange. I don't know what the unit `size: 221` is referencing. I wonder if we could figure out what unit this is and add it to the discovery info?

[the remark this answers is not in the document]
Comment (djwhatle) [reply pkg/apis/migration/v1alpha1/migplan_types.go]: Also, I guess it's expected for us to not show `storageClass` in the discovery list when none exists? Makes sense, that's what I'm seeing here.

[the remark this answers is not in the document]
Comment (jortel) [reply pkg/apis/migration/v1alpha1/migplan_types.go]: Stepped through the debugger and turns out the `PersistentVolumeSpec.Size()` seems to calculate a display width related to `oc get pv` with and NOT the actual storage size.  

Glad you questioned this.  Removed.

**Pull request merged**